### PR TITLE
Adding Loss, LossFunctionWrapper, MeanSquaredError classes.

### PR DESCRIPTION
**Pull request by pavithrasv**
### Summary
Adding Loss, LossFunctionWrapper, MeanSquaredError classes and test cases for testing MeanSquaredError loss class. These classes cannot be used with Keras training yet.

### Related Issues
None

### PR Overview

- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: keras/losses.py [hunk 1]
```diff
@@ -4,12 +4,169 @@
...
 
 
+class Loss(object):
+    """Loss base class.
+
+    To be implemented by subclasses:
+    * `call()`: Contains the logic for loss calculation using `y_true`, `y_pred`.
+
+    Example subclass implementation:
+    ```
```
Comment (fchollet) [inline keras/losses.py]: Add: python

File: keras/losses.py [hunk 2]
```diff
@@ -4,12 +4,169 @@
...
+    class MeanSquaredError(Loss):
+    def call(self, y_true, y_pred):
```
Comment (fchollet) [inline keras/losses.py]: Fix indent; use 4 space indent.

File: keras/losses.py [hunk 3]
```diff
@@ -4,12 +4,169 @@
...
+      y_pred = ops.convert_to_tensor(y_pred)
+      y_true = math_ops.cast(y_true, y_pred.dtype)
+      return K.mean(math_ops.square(y_pred - y_true), axis=-1)
+    ```
+
+    Args:
```
Comment (fchollet) [inline keras/losses.py]: Use convention `# Arguments` --

- start with `#`
- no `:`

This applies to all new docstrings.

File: keras/losses.py [hunk 4]
```diff
@@ -4,12 +4,169 @@
...
+          y_pred: The predicted values.
+        """
+        NotImplementedError('Must be implemented in subclasses.')
+
+
+class LossFunctionWrapper(Loss):
+    """Wraps a loss function in the `Loss` class.
+
+    Args:
+    fn: The loss function to wrap, with signature `fn(y_true, y_pred,
```
Comment (fchollet) [inline keras/losses.py]: Fix indent.

File: keras/losses.py [hunk 5]
```diff
@@ -4,12 +4,169 @@
...
+
+    For example, if `y_true` is [0., 0., 1., 1.] and `y_pred` is [1., 1., 1., 0.]
+    then the mean squared error value is 3/4 (0.75).
+
+    Standalone usage:
+
+    ```python
+    mse = keras.losses.MeanSquaredError()
+    loss = mse([0., 0., 1., 1.], [1., 1., 1., 0.])
+    print('Loss: ', loss.numpy())  # Loss: 0.75
```
Comment (fchollet) [inline keras/losses.py]: `.numpy()` would be TF-specific. Please remove print statement.

File: keras/utils/losses_utils.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+def squeeze_or_expand_dimensions(y_pred, y_true, sample_weight):
+    """Squeeze or expand last dimension if needed.
+
+    1. Squeezes last dim of `y_pred` or `y_true` if their rank differs by 1.
+    2. Squeezes or expands last dim of `sample_weight` if its rank differs by 1
+    from the new rank of `y_pred`.
+    If `sample_weight` is scalar, it is kept scalar.
+
+    # Arguments:
```
Comment (fchollet) [inline keras/utils/losses_utils.py]: No `:`; use 4-space indent.

File: keras/utils/losses_utils.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      y_pred: Predicted values, a `Tensor` of arbitrary dimensions.
+      y_true: Optional label `Tensor` whose dimensions match `y_pred`.
+      sample_weight: Optional weight scalar or `Tensor` whose dimensions match
+        `y_pred`.
+
+    # Returns:
```
Comment (fchollet) [inline keras/utils/losses_utils.py]: No `:`; use 4-space indent.

File: keras/utils/losses_utils.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    return loss
+
+
+def compute_weighted_loss(losses,
+                          sample_weight=None,
+                          reduction=Reduction.SUM_OVER_BATCH_SIZE,
+                          name=None):
+    """Computes the weighted loss.
+
+    # Arguments:
```
Comment (fchollet) [inline keras/utils/losses_utils.py]: No `:`; use 4-space indent.

**Comment by pavithrasv** [comment]
Fixed all the formatting issues. Thank you!

File: keras/losses.py [hunk 9]
```diff
@@ -4,12 +4,173 @@
...
+
+    @abc.abstractmethod
+    def call(self, y_true, y_pred):
+        """Invokes the `Loss` instance.
+
+        # Arguments
+            y_true: Ground truth values, with the same shape as 'y_pred'.
+            y_pred: The predicted values.
+        """
+        NotImplementedError('Must be implemented in subclasses.')
```
Comment (fchollet) [inline keras/losses.py]: `raise`

File: keras/losses.py [hunk 10]
```diff
@@ -4,12 +4,173 @@
...
+
+        # Returns
+            A `Loss` instance.
+        """
+        return cls(**config)
+
+    def get_config(self):
+        return {'reduction': self.reduction, 'name': self.name}
+
+    @abc.abstractmethod
```
Comment (fchollet) [inline keras/losses.py]: Why is this necessary (since we are raising NotImplementedError in any case)?

Comment (pavithrasv) [reply keras/losses.py]: I guess python allows super() calls to abstract methods from subclasses, so we want to avoid that here.
